File: app/src/main/java/com/aayush/filmipedia/util/datasource/factory/CastDataFactory.java
package com.aayush.filmipedia.util.datasource.factory;

import com.aayush.filmipedia.FilmipediaApplication;
import com.aayush.filmipedia.model.Cast;
import com.aayush.filmipedia.util.datasource.CastDataSource;

import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;
import io.reactivex.disposables.CompositeDisposable;

public class CastDataFactory extends DataSource.Factory<Long, Cast> {
    private MutableLiveData<CastDataSource> mutableLiveData;
    private CastDataSource castDataSource;
    private FilmipediaApplication application;
    private CompositeDisposable compositeDisposable;

    private Long personId;

    public CastDataFactory(FilmipediaApplication application,
                           CompositeDisposable compositeDisposable, Long personId) {
        this.application = application;
        this.compositeDisposable = compositeDisposable;
        this.personId = personId;
        mutableLiveData = new MutableLiveData<>();
    }

    @Override
    public DataSource<Long, Cast> create() {
        castDataSource = new CastDataSource(application, compositeDisposable, personId);
        mutableLiveData.postValue(castDataSource);
        return castDataSource;
    }

    public MutableLiveData<CastDataSource> getMutableLiveData() {
        return mutableLiveData;
    }
}

File: app/src/main/java/com/aayush/filmipedia/util/datasource/factory/CastDataFactory.kt
package com.aayush.filmipedia.util.datasource.factory

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import com.aayush.filmipedia.FilmipediaApplication
import com.aayush.filmipedia.model.Cast
import com.aayush.filmipedia.util.datasource.CastDataSource
import io.reactivex.disposables.CompositeDisposable

class CastDataFactory(
    private val application: FilmipediaApplication,
    private val compositeDisposable: CompositeDisposable,
    private val personId: Long?
): DataSource.Factory<Long, Cast>() {
    val mutableLiveData = MutableLiveData<CastDataSource>()
    private lateinit var castDataSource: CastDataSource

    override fun create(): DataSource<Long, Cast> {
        castDataSource = CastDataSource(application, compositeDisposable, personId).apply {  }
        mutableLiveData.postValue(castDataSource)
        return castDataSource
    }
}